core/src/main/java/fj/Equal.java

```diff
@@ -455,6 +442,37 @@ public boolean equal(A a1, A a2) {
     });
   }
 
+  /**
+   * An equal instance for the {@link Zipper} type.
+   *
+   * @param ea Equality across the elements of the zipper.
+   * @return An equal instance for the {@link Zipper} type.
+   */
+  public static <A> Equal<Zipper<A>> zipperEqual(final Equal<A> ea) {
+    Definition<A> eaDef = ea.def;
+    return equalDef((a1, a2) -> {
+      Zipper<A> x1 = a1;
+      Zipper<A> x2 = a2;
+
+      x1.move(0);
```

Comment:
those two `move(0)` statements do nothing?

Reply:
The zipper handed out might not be focus at start. Let me know if you think they are not needed.

Reply:
but `move` is a pure function so you must use the result if you want to do that.
But probably the position of a zipper is important and I'd be in favor of just comparing `lefts()`, `focus()` and `right()`. This is also what the haskell implementations I have seen do.

Reply:
thanks for the pointer. updated